src/collector.rs

```diff
@@ -88,13 +73,115 @@ impl DebugFlags {
     }
 }
 
-pub(crate) struct MarkingCtxt<'mrk> {
-    pub(crate) worklist: GcVec<Block<'mrk>>,
-    pub(crate) stack_start: usize,
+/// Contains data expected to last for the duration of a single collection
+/// cycle.
+struct CollectionCtxt<'a> {
+    /// The allocation cache is a freeze-frame of all block pointers in-use by
+    /// the mutator. It is a sorted vector which allows for fast pointer
+    /// identification using binary search. It is valid up until the drop_phase
+    /// of a collection.
+    allocation_cache: AllocCache<'a>,
+    /// The data segment of an ELF binary contains statics. This section needs
+    /// scanning during marking as it could contain GC roots.
     pub(crate) data_segment_start: usize,
     pub(crate) data_segment_end: usize,
 }
 
+impl<'a> CollectionCtxt<'a> {
+    pub(crate) fn find(&mut self, word: usize) -> Option<Block<'a>> {
+        if !(word >= self.data_segment_end && word <= unsafe { HEAP_TOP }) {
```

Comment:
Why is `HEAP_TOP` surrounded by `unsafe`?

Reply:
It's a mutable static

Reply:
OK.